frontend/appflowy_flutter/lib/ai/service/select_model_bloc.dart

```diff
@@ -1,82 +1,113 @@
   SelectModelBloc({
-    required this.objectId,
-  })  : _aiModelStateNotifier = AIModelStateNotifier(objectId: objectId),
-        super(const SelectModelState()) {
-    _aiModelStateNotifier.init();
-    _aiModelStateNotifier.startListening(
-      onAvailableModelsChanged: (models) {
-        if (!isClosed) {
-          add(SelectModelEvent.didLoadModels(models));
-        }
-      },
-    );
-
+    required AIModelStateNotifier aiModelStateNotifier,
+  })  : _aiModelStateNotifier = aiModelStateNotifier,
+        super(SelectModelState.initial(aiModelStateNotifier)) {
     on<SelectModelEvent>(
-      (event, emit) async {
-        await event.when(
-          selectModel: (AIModelPB model) async {
-            await AIEventUpdateSelectedModel(
+      (event, emit) {
+        event.when(
+          selectModel: (model) {
+            AIEventUpdateSelectedModel(
               UpdateSelectedModelPB(
-                source: objectId,
-                selectedModel: model,
+                source: _aiModelStateNotifier.objectId,
+                selectedModel: model.toPB(),
               ),
             ).send();
 
-            state.availableModels?.freeze();
-            final newAvailableModels = state.availableModels?.rebuild((m) {
-              m.selectedModel = model;
-            });
-
+            emit(state.copyWith(selectedModel: model));
+          },
+          didLoadModels: (models, selectedModel) {
             emit(
-              state.copyWith(
-                availableModels: newAvailableModels,
+              SelectModelState(
+                models: models,
+                selectedModel: selectedModel,
               ),
             );
           },
-          didLoadModels: (AvailableModelsPB models) {
-            emit(state.copyWith(availableModels: models));
-          },
         );
       },
     );
+
+    _aiModelStateNotifier.addListener(
+      onAvailableModelsChanged: _onAvailableModelsChanged,
+    );
   }
 
-  final String objectId;
   final AIModelStateNotifier _aiModelStateNotifier;
 
   @override
   Future<void> close() async {
-    await _aiModelStateNotifier.stop();
+    _aiModelStateNotifier.removeListener(
+      onAvailableModelsChanged: _onAvailableModelsChanged,
+    );
     await super.close();
   }
+
+  void _onAvailableModelsChanged(List<AiModel> models, AiModel? selectedModel) {
+    if (!isClosed) {
+      add(SelectModelEvent.didLoadModels(models, selectedModel));
+    }
+  }
 }
 
 @freezed
 class SelectModelEvent with _$SelectModelEvent {
   const factory SelectModelEvent.selectModel(
-    AIModelPB model,
+    AiModel model,
   ) = _SelectModel;
 
   const factory SelectModelEvent.didLoadModels(
-    AvailableModelsPB models,
+    List<AiModel> models,
+    AiModel? selectedModel,
   ) = _DidLoadModels;
 }
 
 @freezed
 class SelectModelState with _$SelectModelState {
   const factory SelectModelState({
-    AvailableModelsPB? availableModels,
```

Comment:
I prefer use PB directly. We use PB as single source of true for data struct that used in flutter side